soroban-sdk/src/tests/contracttoken.rs

```diff
@@ -0,0 +1,85 @@
+use crate as soroban_sdk;
+use soroban_sdk::{
+    contractclient, contracttype, testutils::AccountId as _, AccountId, Bytes, BytesN, Env,
+};
+
+#[contractclient(name = "TokenClient")]
+pub trait Token {
+    fn init(env: Env, admin: Identifier, metadata: TokenMetadata);
+    fn name(env: Env) -> Bytes;
+}
+
+#[derive(Clone)]
+#[contracttype]
+pub enum Identifier {
+    Account(AccountId),
+}
+
+#[derive(Clone)]
+#[contracttype]
+pub struct TokenMetadata {
+    pub name: Bytes,
+    pub symbol: Bytes,
+    pub decimals: u32,
+}
+
+#[test]
+fn test_register_token() {
```

Comment:
When we re-add `register_stellar_asset_contract`, an you make sure all the tests in this file get added for it to maintain the same level of test coverage for these cases?

Reply:
Hmm I used `invoke_function` in `register_stellar_asset_contract`, which checks if a contract already exists at a specified contractID. What's the reason for writing the contract directly to storage with `register_contract_token`? This is the only test coverage missing.